### PR TITLE
React Scripts and Moment Upgrade

### PR DESCRIPTION
**Pull request by nnti**
## Purpose

Upgraded moment library, and react scripts. The current version of react scripts, 4.0.3, was preventing dependabot from upgrading libraries with vulnerabilities such as node-forge. Current version of react-scripts is 5.0.1. 

#### Linked Issues to Close

Closes: https://github.com/CMSgov/macpro-quickstart-serverless/security/dependabot/2
https://github.com/CMSgov/macpro-quickstart-serverless/security/dependabot/77
https://github.com/CMSgov/macpro-quickstart-serverless/security/dependabot/53

## Approach

Upgrades moment and react scripts library. Remove and replace html library with axios.

#### Notes
Also, webpack < 5 used to include polyfills for node.js core modules by default. This is no longer the case. Because of this, the http node library is replaced with axios.

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

**Review by mdial89f — CHANGES_REQUESTED** [review]
@nnti 
- You'll have to fill out the PR details; right now it looks to be just a blank template.
- I think you mistakenly checked in a __screenshots__ folder.  If so, please remove.

**Comment by mdial89f** [comment]
@13bfrancis or @benjaminpaige 
Could you review the react changes here?

**Review by kenrickveit — CHANGES_REQUESTED** [review]
@nnti a general comment here- it's unclear to me from the `Conversation` what changed for React and why.  I could probably eventually figure it out, but it should be explained in the Convo.  Can you please document it there?

@mdial89f is going to look at this PR and delegate to @13bfrancis on the REACT stuff.

Note you'll need to address the requested changes from @mdial89f and myself and then we can mark those as resolved.

Thanks!

**Comment by mdial89f** [comment]
> > 
> 
> @mdial89f The snapshots directory checked in is intentional.

Why?  Isn't it just a jest output for a given test run?  thanks

**Comment by nnti** [comment]
> > > 
> > 
> > 
> > @mdial89f The snapshots directory checked in is intentional.
> 
> Why? Isn't it just a jest output for a given test run? thanks

The unit test failed for that test so @benjaminpaige modified the output for it to pass. Here is the failed unit test: https://github.com/CMSgov/macpro-quickstart-serverless/runs/6179686647?check_suite_focus=true

**Comment by mdial89f** [comment]
> > > > 
> > > 
> > > 
> > > @mdial89f The snapshots directory checked in is intentional.
> > 
> > 
> > Why? Isn't it just a jest output for a given test run? thanks
> 
> The unit test for that failed so @benjaminpaige modified the output for it to pass. Here is the failed unit test: https://github.com/CMSgov/macpro-quickstart-serverless/runs/6179686647?check_suite_focus=true

OK cool, thanks.  I'm going to resolve my requested changes, and defer to brian/ben for their stamp.  Thanks

**Review by mdial89f — APPROVED** [review]

File: services/ui-src/src/containers/Amendments.js
Comment (benjaminpaige) [inline services/ui-src/src/containers/Amendments.js:3]: im sure that this (axios) is available in the node modules because some other package is using it. But we should either install it explicitly, or just use the fetch api.

File: services/ui-src/patches/@cmsgov+design-system+2.13.0.patch
Comment (benjaminpaige) [inline services/ui-src/patches/@cmsgov+design-system+2.13.0.patch:13]: i know that we did this to get rid of some console errors from the @cmsgov/design-system package. But after looking at the package it looks as though we should be able to just update it to the latest version (3.3.0) and then remove this file and the postinstall script.

**Review by benjaminpaige — COMMENTED** [review]
Couple adjustments suggested, let me know if i can help explain anything about the comments.

File: services/ui-src/src/containers/Amendments.js
Comment (nnti) [reply services/ui-src/src/containers/Amendments.js:3]: ok I'll use fetch as oppose to axios

File: services/ui-src/patches/@cmsgov+design-system+2.13.0.patch
Comment (nnti) [reply services/ui-src/patches/@cmsgov+design-system+2.13.0.patch:13]: yea I agree

File: services/ui-src/src/tests/Header.test.js
Comment (benjaminpaige) [inline services/ui-src/src/tests/Header.test.js:13]: its unfortunate... but we do need this here if we want to capture a snapshot of the header comp. the problem originates from a misconfiguration in the UsaBanner module in the cms design system package. I will log an issue with them

Comment (benjaminpaige) [inline services/ui-src/src/tests/Header.test.js:13]: its unfortunate... but we do need this here if we want to capture a snapshot of the header comp. the problem originates from a misconfiguration in the UsaBanner module in the cms design system package. I will log an issue with them

Comment (nnti) [reply services/ui-src/src/tests/Header.test.js:13]: ok sounds good, thanks for taking a look at this.

**Review by benjaminpaige — APPROVED** [review]
as far as getting rid of 'http' after updating to rs5 to resolve build issues and updating cmsgov/degisn-system package to fix console warnings this code looks good.

**Comment by kenrickveit** [comment]
as mentioned in chat, @nnti the top-level package.json appears to have changes due to the changes in the top level yarn.lock, but it's not checked in.

**Review by kenrickveit — APPROVED** [review]
@nnti thanks for adding additional info on the convo. It's now much more clear what is changing and why.

On the massively different yarn.lock files, I discussed with Nathan and found he re-generated the yarn.lock for both `services/ui-src` and the yarn.lock in the root.  This explains the changes to the latter without a change to package.json.  I'm fine with this, as it will move our dependencies forward as far as possible, some of which might have been quite dated.  It's possible Nathan may need to re-patch some dependabot findings that were fixed by pinning in yarn.lock, but it's probably easiest to determine this after a merge.

**Pull request merged**
